src/EventTeachingService.cpp

```diff
@@ -371,70 +378,79 @@ Processed EventTeachingService::handleLearnEvent(VlcbMessage *msg, unsigned int
         controller->sendGRSP(OPC_EVLRN, getServiceID(), CMDERR_INV_EV_IDX);
         return PROCESSED;
       }
-      // Is this a produced event that we know about?
-      // Search the events table by evindex = 1 for a value match with evval.
-      if ((evindex == 1) && (evval > 0))
+    
+      // Is this an existing event? 
       {
-        index = module_config->findExistingEventByEv(evindex, evval);
-        if (index < module_config->EE_MAX_EVENTS)
+        //DEBUG_SERIAL << F("ets> searching for existing event to update") << endl;
+        index = module_config->findExistingEvent(nn, en);
+
+        // not found in event table
+        if (index >= module_config->EE_MAX_EVENTS)
         {
-          module_config->writeEvent(index, &msg->data[1]);
-          // no need to write eventEV as, by definition, it hasn't changed
-          // recreate event hash table entry
-          module_config->updateEvHashEntry(index);
-          return PROCESSED;
+          // Are we a producer and is this a produced event to update?
+          // Wait timeOut seconds to see if producer button pressed.
+          if ((controller->getParam(PAR_FLAGS) & PF_PRODUCER) && (evindex ==1))       
+          {
+            //DEBUG_SERIAL << F("ets> Update candidate") << endl;
+            unsigned long timeStart = millis();
+            bool terminate = false;
+            while ((timeOut > millis() - timeStart) && !terminate)
+            {
+              index = (byte)(*checkProduced)();
+              if (index < module_config->EE_PRODUCED_EVENTS)
+              {
+                // DEBUG_SERIAL << F("ets> Update index ") << index << endl;
+                terminate = true;
```

Comment:
You can use "break;" here and get rid of the variable "terminate".

Reply:
Agreed.  I'll change it.